src/js/angular/core/directives/languageselector/language-selector.directive.js

```diff
@@ -31,12 +33,44 @@ function languageSelector($translate, LocalStorageAdapter, LSKeys, $languageServ
             // translated they will be translated in English
             $translate.fallbackLanguage('en');
 
+            function requiresReload(tab) {
+                return tab.drawnPlugin === 'charts' || tab.drawnPlugin === 'pivot-table-plugin';
+            }
+
             $scope.changeLanguage = function (lang) {
-                $scope.selectedLang = lang
-                $translate.use(lang.key);
-                LocalStorageAdapter.set(LSKeys.PREFERRED_LANG, lang.key);
-                $languageService.setLanguage($scope.selectedLang.key);
-                $scope.$broadcast('language-changed', {locale: lang.key});
+                const yasgui = YasguiComponentDirectiveUtil.getOntotextYasguiElement(QUERY_EDITOR_ID);
+                const promise = yasgui ? yasgui.yasguiComponent.getActivePluginTab() : Promise.resolve(null);
```

Comment:
Should you do `Promise.reject()` if yasgui is not found on the page? This way you can omit some of the checks you do in the handler below.

Reply:
OK. I made some changes to this.

Reply:
Why is this check here? I think it would be better placed in the yasgui-component.directive. My suggestion is to fire an event via EventEmitter here, and in the yasgui directive, cancel it if necessary. This approach was used when a repository is changed.